chore: consolidating spinner color and refactoring institutional qr code modal

georgewrmarshall opened the pull request:
## **Description**
Breaking out parts of https://github.com/MetaMask/metamask-extension/pull/24971 for easier review. This PR replaces deprecated colors in favor of the most updated design system colors. I also noticed that this particular component could benefit from some layout and overflow updates. 

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25011?quickstart=1)

## **Related issues**

Part of: https://github.com/MetaMask/metamask-extension/issues/24964

## **Manual testing steps**

1. Go to the storybook build of this page
2. Search for `Spinner` and `QRCodeModal`
3. Check colors have been updated properly in light and dark mode

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/d745a119-f42d-429e-9200-d69b5a1776d1

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/282e4460-798a-4841-bad5-242373f7f89b

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

georgewrmarshall committed:
chore: refactoring institutional qr code modal

georgewrmarshall commented on ui/components/institutional/institutional-components.scss:
```diff
@@ -6,3 +6,4 @@
 **/
 @import 'jwt-dropdown/jwt-dropdown';
 @import 'note-to-trader/index';
+@import 'qr-code-modal/qr-code-modal';
```
importing stylesheet to main index styles file

georgewrmarshall commented on ui/components/institutional/qr-code-modal/qr-code-modal.scss:
```diff
@@ -0,0 +1,5 @@
+.institutional-qr-code-modal {
+  &__qr-code {
+    background-color: var(--qr-code-white-background);
```
Moving inline styles to custom classname

georgewrmarshall commented on ui/components/institutional/qr-code-modal/qr-code-modal.stories.tsx:
```diff
@@ -19,7 +19,7 @@ const testData = {
 
 const meta: Meta<typeof QRCodeModal> = {
-  title: 'Components/QRCodeModal',
+  title: 'Components/Institutional/QRCodeModal',
```
Fixing storybook title

### Before

<img width="230" alt="Screenshot 2024-06-03 at 1 54 11 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/ce341811-c2fa-4d68-8a6c-478e5f8f6d14">


### After

<img width="261" alt="Screenshot 2024-06-03 at 1 43 26 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/2826a0c2-fa25-45d5-a008-1596ea8d8db7">

georgewrmarshall commented on ui/components/institutional/qr-code-modal/qr-code-modal.tsx:
```diff
@@ -198,63 +203,61 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
-        {qrCodeValue ? (
-          <Box
-            data-testid="qr-code-visible"
-            style={{
-              padding: 20,
-              backgroundColor: 'var(--qr-code-white-background)',
-              display: 'flex',
-              alignItems: 'center',
-              flexDirection: 'column',
-            }}
```
Removing inlines styles in favor of style utility props and custom classname

georgewrmarshall commented on ui/components/institutional/qr-code-modal/qr-code-modal.tsx:
```diff
@@ -198,63 +203,61 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
-              flexDirection: 'column',
-            }}
+
+        <ModalBody>
```
ModalBody handles overflow using a scroll container

georgewrmarshall commented on ui/components/institutional/qr-code-modal/qr-code-modal.tsx:
```diff
@@ -198,63 +203,61 @@ const QRCodeModal: React.FC<QRCodeModalProps> = ({
+              <QRCode value={qrCodeValue} size={270} />
+            </Box>
+          ) : (
+            <Spinner />
```
We can remove the color as it now defaults to the correct icon-muted color

georgewrmarshall commented on ui/components/ui/spinner/spinner.component.js:
```diff
@@ -1,7 +1,7 @@
 
-const Spinner = ({ className = '', color = 'var(--color-text-default)' }) => {
+const Spinner = ({ className = '', color = 'var(--color-icon-muted)' }) => {
   return (
```
### Before


https://github.com/MetaMask/metamask-extension/assets/8112138/e6f07090-c2c9-4a98-929e-5c7879d840ee

### After


https://github.com/MetaMask/metamask-extension/assets/8112138/e3babf7f-3287-49a5-a01a-f6e12db6ccb5

georgewrmarshall commented on ui/components/ui/spinner/spinner.stories.js:
```diff
@@ -13,7 +13,7 @@ export default {
   },
   args: {
-    color: 'var(--color-warning-default)',
+    color: 'var(--color-icon-muted)',
```
Same as above

brianacnguyen reviewed (dismissed):
approved on ds side

albertolive commented:
Hi @georgewrmarshall,

The refactoring looks great! I really appreciate your work on this.

I have one small request: Could you please reduce the size of the loader? It currently appears quite large, and adjusting it to a smaller size would be perfect.

georgewrmarshall committed:
Merge branch 'develop' into update-qr-code-modal

georgewrmarshall committed:
chore: updating size of the spinner

georgewrmarshall commented on ui/components/institutional/qr-code-modal/qr-code-modal.scss:
```diff
@@ -0,0 +1,9 @@
+    background-color: var(--qr-code-white-background);
+  }
+
+  &__spinner {
+    width: 100px;
+  }
```
Reducing the size of the spinner

georgewrmarshall commented:
> Hi @georgewrmarshall,
> 
> The refactoring looks great! I really appreciate your work on this.
> 
> I have one small request: Could you please reduce the size of the loader? It currently appears quite large, and adjusting it to a smaller size would be perfect.

Not a problem! Updated and updated screencasts in description @albertolive
